reframe/utility/os_ext.py

```diff
@@ -105,6 +105,10 @@ def copytree_virtual(src, dst, file_links=[],
     src = os.path.abspath(src)
     dst = os.path.abspath(dst)
 
+    if src == os.path.commonpath([src, dst]):
+        raise ValueError("copytree_virtual() failed: "
+                         "`%s' is a parent directory of `%s'" % (src, dst))
```

Comment:
The problem is not in this function, but in the `copytree()` a bit above. This version of `copytree()` differs from the `shutil.copytree()` in that it deletes the destination first, if it exists. Also, I don't know if your fix here is the correct behaviour, because it's pretty much valid and well-defined to do `cp -r . ..`. So, we should get back to the scenario of the bug and see what would be the intended and/or the expected behaviour.

Reply:
Yes but is recursively copying `..` to `.`  valid? i.e. `cp -r .. .`?

Reply:
If you create a directory hierarchy`foo/bar`, `cd` into `foo/bar` and try `cp -r .. .` you get the following error:
` cp: cannot copy a directory, '..', into itself, '.'`
If the dir has lots of files, the whole directory structure will be copied before throwing the error. Therefore I believe that we should somehow not allow that to happen by considering it an error before trying to perform the copying of the tree.

Reply:
After some investigation, your solution seems to be correct. However, it should be put inside the `copytree()` function and not the `copytree_virtual()`. The error message should be a bit different as well: ``"cannot copy recursively the parent directory `%s' into one of its descendants `%s'"``.